File: Smarter Lock/Payload/UnlockPayload.cpp
//
//  UnlockPayload.m
//  Smarter Lock
//
//  Created by Yuwei Huang on 15/4/22.
//  Copyright (c) 2015年 CSE481. All rights reserved.
//

#import "UnlockPayload.h"

static UnlockPayload* DefaultUnlockPayload = nullptr;

int UnlockPayload::serialize(uint8_t* output) const {
	return 0;
}

uint32_t UnlockPayload::length() const {
    return 0;
}

uint32_t UnlockPayload::type() const {
    return Type::UNLOCK;
}

Payload* UnlockPayload::DefaultPayload() {
    if (!DefaultUnlockPayload)
        DefaultUnlockPayload = new UnlockPayload();
    
    return DefaultUnlockPayload;
}

File: Smarter Lock/UnlockPayload.m
//
//  UnlockPayload.m
//  Smarter Lock
//
//  Created by Yuwei Huang on 15/4/22.
//  Copyright (c) 2015年 CSE481. All rights reserved.
//

#import "UnlockPayload.h"

@implementation UnlockPayload
UnlockPayload* DefaultPayload = nil;

-(const char*)serialize {
    return nil;
}

-(u_int32_t)length {
    return 0;
}

-(u_int32_t)type {
    return 1;
}

-(id)init {
    return self;
}

+(id)defaultPayload {
    if (!DefaultPayload)
        DefaultPayload = [[UnlockPayload alloc] init];
    
    return DefaultPayload;
}
@end
